Sources/SwiftDocC/Model/Rendering/References/ExternalImageReference.swift

```diff
@@ -0,0 +1,102 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+
+/// A reference to an extern image.
+public struct ExternalImageReference: RenderReference {
```

Comment:
Would it be possible to use `ImageReference` for both local and external images?

At a glance this looks like an `ImageReference` without a `DataAsset` value. 

I'm wondering if making the `asset` optional (which would have to be made on all MediaReference types) or if passing an empty `DataAsset` (or another value signifying an external asset) could work.

Reply:
Yes, I think the same at first. But later I thought hacking the current ImageReference may require changing more code. I'll update the implementation to use ImageReference if this is the original design.

Reply:
Is `fallbackAssetResolvers` the right direction for solving this problem?

Currently fallbackAssetResolvers is an empty array. Following your suggestion, I think I should create a class which conforms to `FallbackAssetResolver` and register it at the proper time.

https://github.com/apple/swift-docc/blob/0d61dc5550c9bdb04fd1579144679e7d3287f640/Sources/SwiftDocC/Infrastructure/DocumentationContext.swift#L2541-L2562

Reply:
> Is `fallbackAssetResolvers` the right direction for solving this problem?

A fallback resolver will behave correctly but it's a public property and it's possible to customize. So it could happen that a developer would unintentionally disable this behavior if they _assigned_ the `fallbackAssetResolvers` property instead of _appending_ to it. 

We haven't documented the default values for any of the public resolver related properties nor have we documented how developers should interact with them. Personally I feel that external image support should be considered a built in feature that can't be disabled but I don't hold that opinion strongly.

Reply:
Yes, I found that after I set a fallbackAssetResolvers manually with my `ExternalMediaResolver`, the `testConvertPageWithLinkResolving` test will fail.
```
if let linkResolvingServer = linkResolvingServer {
    let resolver = try OutOfProcessReferenceResolver(
        bundleIdentifier: request.bundleInfo.identifier,
        server: linkResolvingServer,
        convertRequestIdentifier: messageIdentifier
    )
    
    context.fallbackReferenceResolvers[request.bundleInfo.identifier] = resolver
    context.fallbackAssetResolvers[request.bundleInfo.identifier] = resolver
    context.externalSymbolResolver = resolver
}
```
So should we add a new `externalAssetResolvers` property or just hardcode it at someplace to just ignore the https and http external image(Which is really a bad idea). How can we determine if a given schema is external accessible or something that a reserved  like `docc://image.png` and `docs-media:///path/to/image.png`